### ISSUE TITLE
Docker startup script does not use local config

### ISSUE THREAD

**Issue by ToshY** [crowdsecurity/crowdsec#2121]
### What happened?

I have setup Crowdsec and a MySQL database in my docker compose configuration (see reproduction steps), and when upgrading from `v1.4.2-debian` to `v1.4.3-debian`, I keep getting the following error message in my Crowdsec container:

```
chown: cannot access '/var/lib/crowdsec/data/crowdsec.db': No such file or directory
```

After extensive debugging, I've actually found out that starting from `v1.4.3`, the `docker_start.sh` had been changed to include `shopt -s inherit_errexit`, which results on exit when something fails in the script, which in this case is `chown` operation. Making this stricter is fine and not the issue, but it made me realise that it only read `/etc/crowdsec/config.yaml` and doesn't do anything with possible local configuration files that are present, e.g. `/etc/crowdsec/config.yaml.local`.

#### TLDR

The `docker_start.sh` does not take into consideration `config.yaml.local`, unlike stated in the [documentation](https://github.com/crowdsecurity/crowdsec/blob/master/docker/README.md#full-configuration) (which was changed in [PR 2115](https://github.com/crowdsecurity/crowdsec/pull/2115)).

### What did you expect to happen?

Local config gets merged with the standard config.

* Check if `config.yaml.local` exists, if so merge it with the standard config (local values overriding standard config).
* Use [yq](https://mikefarah.gitbook.io/yq/operators/multiply-merge#merge-two-files-together) for merging.

### How can we reproduce it (as minimally and precisely as possible)?

`./crowdsec/Dockerfile`
```Dockerfile
ARG CROWDSEC_IMAGE_VERSION

FROM crowdsecurity/crowdsec:${CROWDSEC_IMAGE_VERSION}

COPY ./config/config.yaml.local \
     /etc/crowdsec/
```

`./crowdsec/config/config.yaml.local`
```yaml
db_config:
  log_level: info
  type: ${DATABASE_TYPE}
  user: ${DATABASE_USER}
  password: ${DATABASE_USER_PASSWORD}
  db_name: ${DATABASE_NAME}
  host: ${DATABASE_HOST}
  port: ${DATABASE_PORT}
api:
  server:
    log_level: info
    listen_uri: 0.0.0.0:8081
    trusted_ips:
      - 172.17.0.0/16
```

`./.env`
```dotenv
CROWDSEC_IMAGE_VERSION="v1.4.3"
MYSQL_HOST="mysql"
MYSQL_PORT="3306"
MYSQL_DATABASE="crowdsec"
MYSQL_ROOT_PASSWORD="<omitted>"
MYSQL_USER="<omitted>"
MYSQL_PASSWORD="<omitted>"
```

`./docker-compose.yml`
```yaml
version: '3.9'

services:
  crowdsec:
    build:
      context: data/crowdsec
      dockerfile: Dockerfile
      args:
        CROWDSEC_IMAGE_VERSION: ${CROWDSEC_IMAGE_VERSION}
    depends_on:
      - mysql
    environment:
      GID: ${GID:-1001}
      DATABASE_TYPE: ${MYSQL_HOST}
      DATABASE_USER: ${MYSQL_USER}
      DATABASE_USER_PASSWORD: ${MYSQL_PASSWORD}
      DATABASE_NAME: ${MYSQL_DATABASE}
      DATABASE_HOST: ${MYSQL_HOST}
      DATABASE_PORT: ${MYSQL_PORT}
    volumes:
      - crowdsec:/var/lib/crowdsec/data/
      - crowdsec_config:/etc/crowdsec/
      - /var/log/kern.log:/log/kern.log:ro
      - /var/log/auth.log:/log/auth.log:ro
      - proxy_logs:/log/traefik/:ro
    networks:
      - crowdsec

  mysql:
    #...

  # volumes, networks and other unimportant stuff omitted
```

### Anything else we need to know?

Question originally asked in Discord Support channel:

 `Docker Compose | Upgrade 1.4.2 to 1.4.3 fails with chown cannot access crowdsec.db` (id: 1084248252901228594).

### Crowdsec version

<details>
v1.4.3
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
$ uname -a
x86_64
```

</details>

### Enabled collections and parsers

<n/a>

### Acquisition config

<n/a>

### Config show

<n/a - startup fails>

### Prometheus metrics

<n/a>

### Related custom configs versions (if applicable) : notification plugins, custom scenarios, parsers etc.

Logs:

```text
crowdsec-crowdsec-1      | 2023-03-11T22:15:58.664451787Z Regenerate local agent credentials
crowdsec-crowdsec-1      | 2023-03-11T22:15:58.796570140Z time="11-03-2023 22:15:58" level=info msg="Prepending yaml: '/etc/crowdsec/profiles.yaml' with '/etc/crowdsec/profiles.yaml.local'"
crowdsec-crowdsec-1      | 2023-03-11T22:15:59.008973299Z time="11-03-2023 22:15:59" level=info msg="Machine 'localhost' successfully added to the local API"
crowdsec-crowdsec-1      | 2023-03-11T22:15:59.009881957Z time="11-03-2023 22:15:59" level=info msg="API credentials dumped to '/etc/crowdsec/local_api_credentials.yaml'"
crowdsec-crowdsec-1      | 2023-03-11T22:15:59.019535056Z Check if lapi needs to register an additional agent
crowdsec-crowdsec-1      | 2023-03-11T22:15:59.162179386Z chown: cannot access '/var/lib/crowdsec/data/crowdsec.db': No such file or directory
```

**Comment by LaurenceJJones** [comment]
If you provide a GID then this issue comes to light

https://github.com/crowdsecurity/crowdsec/blob/b451d190b766d7fb5e64c7f1cedde314b02e1f0c/docker/docker_start.sh#L268-L269

But yes fundamentally, the config helper does not take .local into account.

**Comment by ToshY** [comment]
hey @LaurenceJJones,

Makes sense that it would be unnecessary to pass the `$GID` when I use an external database. But like you also stated, it does not take into consideration the local config, which is why I created the issue.

I think is not that hard to fix if we just check if a local config exists and merge it with the standard config inside the startup script.

I'm willing to give it a shot myself and create a PR for it. Let me know 🙂

**Comment by LaurenceJJones** [comment]
Looking through the script, that seems to be the only place I would say that needs it adding, as the rest are checking if environment is set then update the `/etc/crowdsec/config.yaml` as you shouldn't be mounting this file a per instructions. So maybe something like

```bash
# crowdsec sqlite database permissions
if [ "$GID" != "" ]; then
    if [ ! -e "/etc/crowdsec/config.yaml.local" ] && istrue "$(conf_get '.db_config.type == "sqlite"')"; then
        chown ":$GID" "$(conf_get '.db_config.db_path')"
        echo "sqlite database permissions updated"
    elif istrue "$(conf_get '.db_config.type == "sqlite"' '/etc/crowdsec/config.yaml.local')"; then
        chown ":$GID" "$(conf_get '.db_config.db_path' '/etc/crowdsec/config.yaml.local')"
        echo "sqlite database permissions updated"
    else
        echo "You are providing a GID but you are not using sqlite database."
    fi
fi
```

**Comment by mmetc** [comment]
I think `chmod -f .... && echo ...` should be enough with no ill effect

Regarding .local, the issues are

 - conf_set will always ignore it
 - not sure yq handles the merge the same way crowdsec does
 - `cscli config show -o raw` could be used, but it comes from memory and it's not the original data

I can try with yq but would like more test coverage

**Comment by mmetc** [comment]
not tested

**Comment by ToshY** [comment]
@mmetc I've tested `v1.5.3-rc4-debian` (which includes these changes), but now I have differnt issues when mounting the `/etc/crowdsec/local_api_credentials.yaml`.

<details>
<summary><code>docker-compose.yml</code></summary>

```yaml
version: '3.9'

x-restart-stopped: &restart-stopped
  restart: unless-stopped

x-restart-always: &restart-always
  restart: always

services:
  crowdsec:
    image: crowdsecurity/crowdsec:v1.5.3-rc4-debian
    depends_on:
      - mysql
    environment:
      DEBUG: 'TRUE'
      LEVEL_TRACE: 'TRUE'
      COLLECTIONS: ${CROWDSEC_COLLECTIONS}
      PARSERS: ${CROWDSEC_PARSERS}
      POSTOVERFLOWS: ${CROWDSEC_POSTOVERFLOWS}
      DATABASE_TYPE: ${MYSQL_HOST}
      DATABASE_USER: ${MYSQL_USER}
      DATABASE_USER_PASSWORD: ${MYSQL_PASSWORD}
      DATABASE_NAME: ${MYSQL_DATABASE}
      DATABASE_HOST: ${MYSQL_HOST}
      DATABASE_PORT: ${MYSQL_PORT}
      BOUNCER_KEY_TRAEFIK: ${CROWDSEC_BOUNCER_API_KEY}
    volumes:
      - ./data/crowdsec/config/local_api_credentials.yaml:/etc/crowdsec/local_api_credentials.yaml
      - ./data/crowdsec/config/online_api_credentials.yaml:/etc/crowdsec/online_api_credentials.yaml
      - ./data/crowdsec/config/acquis.d:/etc/crowdsec/acquis.d
      - ./data/crowdsec/config/config.yaml.local:/etc/crowdsec/config.yaml.local
      - ./data/crowdsec/config/profiles.yaml.local:/etc/crowdsec/profiles.yaml.local
      - ./data/crowdsec/notifications/discord.yaml:/etc/crowdsec/notifications/discord.yaml
      - ./data/crowdsec/parsers/s02-enrich/user-whitelist.yaml:/etc/crowdsec/parsers/s02-enrich/user-whitelist.yaml
      - ./data/crowdsec/postoverflows/s01-whitelist/fqdn-whitelists.yaml:/etc/crowdsec/postoverflows/s01-whitelist/fqdn-whitelists.yaml
      - crowdsec_datax:/var/lib/crowdsec/data
      - /var/log:/var/log:ro
      - proxy_logs:/log/traefik:ro
    ports:
      - "8282:8080"
    labels:
      traefik.enable: true
      traefik.http.routers.crowdsec.service: crowdsec
      traefik.http.services.crowdsec.loadbalancer.server.port: 8080
    networks:
      - crowdsec
      - proxy

  bouncer:
    image: fbonalair/traefik-crowdsec-bouncer:${CROWDSEC_TRAEFIK_BOUNCER_IMAGE_VERSION}
    <<: [*restart-stopped]
    depends_on:
      - crowdsec
    environment:
      PORT: 8082
      CROWDSEC_AGENT_HOST: ${CROWDSEC_BOUNCER_AGENT_HOST}
      CROWDSEC_BOUNCER_API_KEY: ${CROWDSEC_BOUNCER_API_KEY}
      TRUSTED_PROXIES: ${CROWDSEC_BOUNCER_TRUSTED_PROXIES}
      GIN_MODE: ${CROWDSEC_BOUNCER_GIN_MODE}
    networks:
      - crowdsec
      - proxy

  mysql:
    build:
      context: data/mysql
      dockerfile: Dockerfile
      args:
        MYSQL_IMAGE_VERSION: ${MYSQL_IMAGE_VERSION}
    <<: [*restart-always]
    cap_add:
      - SYS_NICE
    security_opt:
      - no-new-privileges:true
    logging:
      options:
        max-size: "50m"
        max-file: "5"
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "127.0.0.1", "--silent"]
      interval: 5s
      timeout: 3s
      retries: 2
      start_period: 0s
    environment:
      MYSQL_DATABASE: ${MYSQL_DATABASE}
      MYSQL_USER: ${MYSQL_USER}
      MYSQL_PASSWORD: ${MYSQL_PASSWORD}
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
    volumes:
      - mysqly:/var/lib/mysql
      - ./data/mysql/init:/docker-entrypoint-initdb.d
    networks:
      - crowdsec

  wait:
    image: waisbrot/wait
    environment:
      TARGETS: mysql:3306 crowdsec:8080 bouncer:8082
      TIMEOUT: 60
    networks:
      - crowdsec

volumes:
  crowdsec_datax:
    driver: local
  mysqly:
    driver: local
  proxy_logs:
    external: true

networks:
  crowdsec:
    driver: bridge
  proxy:
    external: true
```

</details>

<details>
<summary><code>/etc/crowdsec/local_api_credentials.yaml</code></summary>

```text
url: http://0.0.0.0:8080
login: localhost
password: xxx
```

</details>

<details>
<summary><code>/etc/crowdsec/config.yaml.local</code></summary>

```text
db_config:
  log_level: info
  type: ${DATABASE_TYPE}
  user: ${DATABASE_USER}
  password: ${DATABASE_USER_PASSWORD}
  db_name: ${DATABASE_NAME}
  host: ${DATABASE_HOST}
  port: ${DATABASE_PORT}
api:
  client:
    insecure_skip_verify: false
    credentials_path: /etc/crowdsec/local_api_credentials.yaml
  server:
    enable: true
    log_level: info
    listen_uri: 0.0.0.0:8080
    trusted_ips:
      - 10.0.0.0/8
      - 172.16.0.0/12
      - 192.168.0.0/16
```

</details>

<details open>
<summary><code>docker compose logs -ft</code></summary>

```text
Attaching to crowdsec-crowdsec-1, crowdsec-mysql-1
crowdsec-crowdsec-1  | + export 'PS4=+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
crowdsec-crowdsec-1  | + PS4='+(${BASH_SOURCE}:${LINENO}): ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
crowdsec-crowdsec-1  | +(docker_start.sh:30): istrue ''
crowdsec-crowdsec-1  | +(docker_start.sh:11): istrue(): case "$(echo "$1" | tr '[:upper:]' '[:lower:]')" in
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): echo ''
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): tr '[:upper:]' '[:lower:]'
crowdsec-crowdsec-1  | +(docker_start.sh:13): istrue(): return 1
crowdsec-crowdsec-1  | +(docker_start.sh:36): export CONFIG_FILE=/etc/crowdsec/config.yaml
crowdsec-crowdsec-1  | +(docker_start.sh:36): CONFIG_FILE=/etc/crowdsec/config.yaml
crowdsec-crowdsec-1  | +(docker_start.sh:37): export CUSTOM_HOSTNAME=localhost
crowdsec-crowdsec-1  | +(docker_start.sh:37): CUSTOM_HOSTNAME=localhost
crowdsec-crowdsec-1  | +(docker_start.sh:146): '[' -n '' ']'
crowdsec-crowdsec-1  | +(docker_start.sh:146): '[' -n '' ']'
crowdsec-crowdsec-1  | +(docker_start.sh:157): for geodb in GeoLite2-ASN.mmdb GeoLite2-City.mmdb
crowdsec-crowdsec-1  | +(docker_start.sh:162): '[' '!' -e /var/lib/crowdsec/data/GeoLite2-ASN.mmdb ']'
crowdsec-crowdsec-1  | +(docker_start.sh:157): for geodb in GeoLite2-ASN.mmdb GeoLite2-City.mmdb
crowdsec-crowdsec-1  | +(docker_start.sh:162): '[' '!' -e /var/lib/crowdsec/data/GeoLite2-City.mmdb ']'
crowdsec-crowdsec-1  | +(docker_start.sh:169): '[' '!' -e /etc/crowdsec/local_api_credentials.yaml ']'
crowdsec-crowdsec-1  | +(docker_start.sh:182): istrue ''
crowdsec-crowdsec-1  | +(docker_start.sh:11): istrue(): case "$(echo "$1" | tr '[:upper:]' '[:lower:]')" in
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): echo ''
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): tr '[:upper:]' '[:lower:]'
crowdsec-crowdsec-1  | +(docker_start.sh:13): istrue(): return 1
crowdsec-crowdsec-1  | +(docker_start.sh:184): '[' -n '' ']'
crowdsec-crowdsec-1  | ++(docker_start.sh:188): conf_get .api.client.credentials_path
crowdsec-crowdsec-1  | ++(docker_start.sh:56): conf_get(): '[' 1 -ge 2 ']'
crowdsec-crowdsec-1  | ++(docker_start.sh:59): conf_get(): cscli config show-yaml
crowdsec-crowdsec-1  | ++(docker_start.sh:50): cscli(): command cscli -c /etc/crowdsec/config.yaml config show-yaml
crowdsec-crowdsec-1  | ++(docker_start.sh:59): conf_get(): yq e .api.client.credentials_path
crowdsec-crowdsec-1  | time="13-08-2023 19:45:32" level=fatal msg="while reading yaml file: open /etc/crowdsec/config.yaml: no such file or directory"
crowdsec-crowdsec-1  | +(docker_start.sh:188): lapi_credentials_path=null
crowdsec-crowdsec-1  | +(docker_start.sh:190): isfalse ''
crowdsec-crowdsec-1  | +(docker_start.sh:18): isfalse(): istrue ''
crowdsec-crowdsec-1  | +(docker_start.sh:11): istrue(): case "$(echo "$1" | tr '[:upper:]' '[:lower:]')" in
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): tr '[:upper:]' '[:lower:]'
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): echo ''
crowdsec-crowdsec-1  | +(docker_start.sh:13): istrue(): return 1
crowdsec-crowdsec-1  | +(docker_start.sh:21): isfalse(): return 0
crowdsec-crowdsec-1  | +(docker_start.sh:193): isfalse ''
crowdsec-crowdsec-1  | +(docker_start.sh:18): isfalse(): istrue ''
crowdsec-crowdsec-1  | +(docker_start.sh:11): istrue(): case "$(echo "$1" | tr '[:upper:]' '[:lower:]')" in
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): echo ''
crowdsec-crowdsec-1  | ++(docker_start.sh:11): istrue(): tr '[:upper:]' '[:lower:]'
crowdsec-crowdsec-1  | +(docker_start.sh:13): istrue(): return 1
crowdsec-crowdsec-1  | +(docker_start.sh:21): isfalse(): return 0
crowdsec-crowdsec-1  | +(docker_start.sh:194): yq -e '.login==strenv(CUSTOM_HOSTNAME)' null
crowdsec-crowdsec-1  | Error: open null: no such file or directory
crowdsec-crowdsec-1  | +(docker_start.sh:197): echo 'Generate local agent credentials'
crowdsec-crowdsec-1  | +(docker_start.sh:200): cscli machines delete localhost
crowdsec-crowdsec-1  | Generate local agent credentials
crowdsec-crowdsec-1  | +(docker_start.sh:200): true
crowdsec-crowdsec-1  | +(docker_start.sh:201): cscli machines add localhost --auto
crowdsec-crowdsec-1  | +(docker_start.sh:50): cscli(): command cscli -c /etc/crowdsec/config.yaml machines add localhost --auto
crowdsec-crowdsec-1  | time="13-08-2023 19:45:32" level=fatal msg="while reading yaml file: open /etc/crowdsec/config.yaml: no such file or directory"
```

</details>

This seems related to changes made in https://github.com/crowdsecurity/crowdsec/commit/f39fbf07fac8a545f2d6fa367014456a7a0f96c6

**Comment by ToshY** [comment]
> @mmetc @LaurenceJJones I've tested `v1.5.3-rc4-debian` (which includes these changes), but now I have different issues when mounting the `/etc/crowdsec/local_api_credentials.yaml`.

@mmetc @LaurenceJJones  Should I make a separate bug report for this?

**Comment by mmetc** [comment]
Yes please, we'll have a look tomorrow
